core/domain/html_cleaner.py

```diff
@@ -405,7 +450,38 @@ def convert_to_text_angular(html_data):
     # html strings they are stored as <br>. Since both of these
     # should match and <br> and <br/> have same working,
     # so the tag has to be replaced in this way.
-    return str(soup).replace('<br/>', '<br>')
+    return unicode(soup).replace('<br/>', '<br>')
+
+
+def tag_content_to_textangular(html_data):
+    """This functions converts content with tabs and collapsible
+    components to textangular format.
+
+    Args:
+        html_data: str. The HTML string whose content is to be converted.
+
+    Returns:
+        str. The HTML string with converted content within tag.
+    """
+    soup = bs4.BeautifulSoup(html_data.encode('utf-8'), 'html.parser')
+
+    for collapsible in soup.findAll('oppia-noninteractive-collapsible'):
+        content_html = html_unescaper(collapsible['content-with-value'])
+        # This is done to ensure that &quot; is not escaped as &quot;
+        # is not escaped as &amp;amp;quot;.
+        collapsible['content-with-value'] = '&quot;' + html_escaper(
+            convert_to_text_angular(content_html[1:-1])) + '&quot;'
+
+    for tabs in soup.findAll('oppia-noninteractive-tabs'):
+        tab_content_json = html_unescaper(tabs['tab_contents-with-value'])
+        tab_content_list = json.loads(tab_content_json)
+        for index, tab_content in enumerate(tab_content_list):
+            tab_content_list[index]['content'] = convert_to_text_angular(
+                tab_content['content'])
+        tabs['tab_contents-with-value'] = html_escaper(
+            json.dumps(tab_content_list))
+
+    return unicode(soup)
```

Comment:
I don't think you need the unicode cast.

Reply:
Answered!